pkg/component/runtime/service.go

```diff
@@ -435,36 +443,64 @@ func (s *serviceRuntime) check(ctx context.Context) error {
 		return ErrOperationSpecUndefined
 	}
 	s.log.Debugf("check if the %s is installed", s.comp.InputSpec.BinaryName)
-	return s.executeServiceCommandImpl(ctx, s.log, s.comp.InputSpec.BinaryPath, s.comp.InputSpec.Spec.Service.Operations.Check, false)
+	return s.executeServiceCommandImpl(ctx, s.log, s.comp.InputSpec.BinaryPath, s.comp.InputSpec.Spec.Service.Operations.Check)
 }
 
 // install executes the service install command
+// The install command executes asynchronously and retries indefinitely until it succeeds.
+// The install command can execute asynchronously because the component will be detected as
+// started and healthy when it checks in, which is already asynchronous from this command.
 func (s *serviceRuntime) install(ctx context.Context) error {
 	if s.comp.InputSpec.Spec.Service.Operations.Install == nil {
 		s.log.Errorf("missing install spec for %s service", s.comp.InputSpec.BinaryName)
 		return ErrOperationSpecUndefined
 	}
 	s.log.Debugf("install %s service", s.comp.InputSpec.BinaryName)
-	return s.executeServiceCommandImpl(ctx, s.log, s.comp.InputSpec.BinaryPath, s.comp.InputSpec.Spec.Service.Operations.Install, true)
+
+	// Install with infinite retries, don't block and execute asynchronously.
+	return s.executeServiceCommandWithRetriesImpl(
+		ctx, s.log, s.comp.InputSpec.BinaryPath, s.comp.InputSpec.Spec.Service.Operations.Install,
+		context.Background(), defaultRetryInitInterval, defaultRetryMaxInterval, false)
```

Comment:
would there be a race condition if the state changes, and there is the go routine running for an hour retrying install?

Reply:
I mean, is it possible that while the retrier install/retrier go routine runs (waits for the next retry( there will be a legitimate state change on the agent that will get Endpoint uninstalled, and then this retrier loop kicks in and installs the Endpoint point? The kind of the race condition, when you have potentially long running (1 hour) go routine that can only be stopped with another install command or 1 hour timeout.

Reply:
Yeah, I see what you mean.  In the scenario you outlined, there will be nothing to cancel the ongoing install command retries so, yeah, I think what you said will happen. 😞 Thanks for bringing it up!

Thinking about how best to fix this.  Perhaps the retrier should cancel _any_ service command for the same service binary before running the new command?